.github/scripts/get-build-artifact.js

```diff
@@ -0,0 +1,52 @@
+/**
+ * Auto-derive artifacts name from successful staging build
+ * 
+ * This script is used in GitHub Actions to automatically find and construct
+ * the artifacts name from a successful staging workflow run that contains
+ * the build-iso-and-end2end-test job.
+ * 
+ * @param {Object} github - GitHub Actions toolkit github object
+ * @param {Object} context - GitHub Actions context object
+ * @param {Object} core - GitHub Actions core object for logging/errors
+ * @returns {Promise<string>} The constructed artifacts name
+ */
+async function getBuildArtifact(github, context, core) {
+  const workflow_id = 'build-iso-and-end2end-test'; // The workflow ID for the build job
+
+  // Get the commit SHA for the tag
+  const tagCommit = context.sha;
+  core.info(`Looking for successful builds for commit: ${tagCommit}`);
+  
+  // Get workflow runs for this repository``
+  const { data: workflowRuns } = await github.rest.actions.listWorkflowRuns({
+    owner: context.repo.owner,
+    repo: context.repo.repo,
+    workflow_id,
+    head_sha: tagCommit,
+    status: 'completed',
+    conclusion: 'success'
+  });
+  
+  // Find the first successful staging workflow run
+  const run = workflowRuns.workflow_runs[0];
+  if (!run) {
+    throw new Error(`No successful end2end workflow run found for commit ${tagCommit}`);
+  }
+  
+  core.info(`Found staging run: ${run.id} with conclusion: ${run.conclusion}`);
+  
+  // Construct artifacts name
+  const commitHash = tagCommit.substring(0, 7); // Use first 7 chars of commit hash
```

Comment:
Shouldn't this be 10 characters ? Previous builds have a 10 character hash.

Examples:
- `github:scality:Zenko:staging-2e37bb5673.build-iso-and-end2end-test.7286`
- `github:scality:Zenko:staging-18ddd8823d.build-iso-and-end2end-test.7297`
- `github:scality:Zenko:staging-24eb369cde.build-iso-and-end2end-test.7295`
- `github:scality:Zenko:staging-6994ac4ee7.build-iso-and-end2end-test.7292`

Reply:
This is the same check which is implemented in the ˋartifacts` action, when promoting an artifact: so it works, but indeed it seems 10 chars are used, I'll check how it is done on upload and/or server side